misc/gam/gam_one_centered_design.stan

```diff
@@ -0,0 +1,177 @@
+////
+// Generalized Additive Model : y ~ s(X)
+// using cubic splines with negative binomial and log-link
+// The spline design matrix is centered and augmented with an intercept
+// Author: Yannick G Spill
+// References:
+// - Generalized Additive Models: an introduction with R, Simon N. Wood, CRC press (2006)
+// - Eilers and Marx, Stat Sci, 1996
+// This work is licensed under a Creative Commons Attribution-ShareAlike 4.0 International License
```

Comment:
That's a doc license, not a code license.  Would you mind 3-clause BSD (aka "new BSD"?)

And you need to indicate who the copyright holder is --- it's usually either you or your employer depending on the arrangement you have with them (many universities, including Columiba, assert copyright ownership of code written by their employees).

And thanks for including the references.